connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -328,6 +335,48 @@ public Set<String> createTopics(NewTopic... topics) {
         return createOrFindTopics(topics).createdTopics();
     }
 
+    /**
+     * Implements a retry logic around creating topic(s) in case it'd fail due to InvalidReplicationFactorException
+     *
+     * @param topicDescription
+     * @param timeoutMs
+     * @param backOffMs
+     * @param time
+     * @return the same as {@link TopicAdmin#createTopics(NewTopic...)}
+     */
+    public Set<String> createTopicsWithRetry(NewTopic topicDescription, long timeoutMs, long backOffMs, Time time) {
```

Comment:
I think it would be a bit more robust to add an overload to {{org.apache.kafka.connect.util.RetryUtil#retryUntilTimeout}} that specifies a set of exceptions or a condition to retry on and use that here.

Reply:
That retrier is constructed for actually retriable exception and in our case the exception is wrapped in another exception and don't really want to touch/refactor/hack this utility method to handle these kind of exceptions too and adding more not-necessarily retryable exceptions to check in the condition where it retries.
This refactor would change the semantics of the retries where it's being used currently.

long story short, I feel this topic creation retry is a bit special and wouldn't been used this way in many places later on.

Reply:
Ok, that's fair.